File: steps/typescript/get-pet.step.ts
// steps/typescript/get-pet.step.ts
import { ApiRouteConfig, Handlers } from 'motia';
import { z } from 'zod';
import { TSStore } from './ts-store';

// Define path parameter schema
const pathParamsSchema = z.object({
  id: z.string().min(1, 'Pet ID is required')
});

export const config: ApiRouteConfig = {
  type: 'api',
  name: 'TsGetPet',
  path: '/ts/pets/:id',
  method: 'GET',
  emits: [],
  flows: ['TsPetManagement']
};

export const handler: Handlers['TsGetPet'] = async (req, { logger }) => {
  try {
    // Validate path parameters
    const { id } = pathParamsSchema.parse(req.pathParams);

    const pet = TSStore.get(id);

    if (!pet) {
      return { status: 404, body: { message: 'Pet not found' } };
    }

    return { status: 200, body: pet };

  } catch (error) {
    if (error instanceof z.ZodError) {
      return {
        status: 400,
        body: {
          message: 'Validation error',
          errors: error.errors
        }
      };
    }

    return {
      status: 500,
      body: { message: 'Internal server error' }
    };
  }
};

File: steps/python/get_pet_step.py
# steps/python/get_pet.step.py
config = { "type":"api", "name":"PyGetPet", "path":"/py/pets/:id", "method":"GET", "emits": [], "flows": ["PyPetManagement"] }

async def handler(req, _ctx=None):
    try:
        import sys
        import os
        sys.path.append(os.path.join(os.path.dirname(__file__), '..'))
        from services import pet_store
    except ImportError:
        return {"status": 500, "body": {"message": "Import error"}}
    pid = req.get("pathParams", {}).get("id")
    pet = pet_store.get(pid)
    return {"status": 200, "body": pet} if pet else {"status": 404, "body": {"message": "Not found"}}
